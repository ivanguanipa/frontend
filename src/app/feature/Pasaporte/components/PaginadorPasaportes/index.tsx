import * as PropTypes from 'prop-types';
import * as React from 'react';

interface PaginadorPasaportesProps {
  cantidadTotalPasaportes: number;
  onClickCambiarPagina: (numeroPagina: number) => void;
}

const PASAPORTES_VISIBLES_POR_PAGINA = 10;

export const PaginadorPasaportes: React.FC<PaginadorPasaportesProps> = ({
  onClickCambiarPagina,
  cantidadTotalPasaportes,
}) => {
  if (cantidadTotalPasaportes <= PASAPORTES_VISIBLES_POR_PAGINA) {
    return null;
  }

  const rango = Array.from(
    Array(
      Math.ceil(cantidadTotalPasaportes / PASAPORTES_VISIBLES_POR_PAGINA)
    ).keys()
  );

  return (
    <nav>
      {rango.map((index) => {
        return (
          <button
            onClick={() => onClickCambiarPagina(index)}
            key={index.toString()}
          >
            {index + 1}
          </button>
        );
      })}
    </nav>
  );
};

PaginadorPasaportes.propTypes = {
  cantidadTotalPasaportes: PropTypes.number.isRequired,
  onClickCambiarPagina: PropTypes.func.isRequired,
};
