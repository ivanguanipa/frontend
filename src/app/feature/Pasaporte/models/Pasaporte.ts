type Nullable<T> = T | null ;
export interface Pasaporte {
    id?: Nullable<number>,
    fullname: string,
    address: string,
    birthdate: string,
    documentId: number,
    applicationDate: string,
    appointmentDate?: Nullable<string>,
    amount?: Nullable<number>,
}