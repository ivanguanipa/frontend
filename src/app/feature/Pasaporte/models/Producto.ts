export interface Producto {
  title: string;
  slug: string;
  body: string;
  createdAt: string;
  updatedAt: string;
}
