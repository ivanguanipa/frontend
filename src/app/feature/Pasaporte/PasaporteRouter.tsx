import * as React from 'react';
import { Route, Switch } from 'react-router-dom';
import { LazyFallback } from 'app/shared/components/LazyFallback';

const MainPage = React.lazy(() => import('./pages/Main'));
const ShowPage = React.lazy(() => import('./pages/Show'));

export const PasaporteRouter = () => (
  <React.Suspense fallback={<LazyFallback />}>
    {/* Layout compartido entre las rutas va aquÃ­ */}
    <Switch>
      <Route path="/pasaportes" exact component={MainPage}></Route>
      <Route path="/pasaportes/show/:id" component={ShowPage}></Route>

    </Switch>
  </React.Suspense>
);
