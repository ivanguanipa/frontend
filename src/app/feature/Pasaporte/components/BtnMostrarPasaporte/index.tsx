import * as PropTypes from 'prop-types';
import * as React from 'react';
import { Button } from 'app/shared/components/Button';
import { Pasaporte } from '../../models/Pasaporte';
import { useHistory } from "react-router-dom";

interface BtnMostrarPasaporteProps {
  pasaporte: Pasaporte;
}

export const BtnMostrarPasaporte: React.FC<BtnMostrarPasaporteProps> = ({
  pasaporte,
}) => {
  let history = useHistory();

  return (
    <Button className={'showPassport'} onClick={() => history.push('/pasaportes/show/'+pasaporte.id)}>
      <span role="img" aria-labelledby="trash">
        Ver
      </span>
    </Button>
  );
};

BtnMostrarPasaporte.propTypes = {
  pasaporte: PropTypes.shape({
    id: PropTypes.number.isRequired,
    fullname: PropTypes.string.isRequired,
    address: PropTypes.string.isRequired,
    birthdate: PropTypes.string.isRequired,
    document_id: PropTypes.number.isRequired,
    application_date: PropTypes.string.isRequired,
    appointment_date: PropTypes.string.isRequired,
    amount: PropTypes.number.isRequired,
  }).isRequired,
};
