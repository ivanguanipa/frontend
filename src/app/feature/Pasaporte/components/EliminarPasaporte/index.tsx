import * as PropTypes from 'prop-types';
import * as React from 'react';
import { Button } from 'app/shared/components/Button';
import { Pasaporte } from '../../models/Pasaporte';

interface BtnEliminarPasaporteProps {
  onEliminar: (pasaporte: Pasaporte) => void;
  pasaporte: Pasaporte;
}

export const BtnEliminarPasaporte: React.FC<BtnEliminarPasaporteProps> = ({
  onEliminar,
  pasaporte,
}) => {

  const handleEliminar = () => {
    if(window.confirm('Esta seguro de eliminar el registro seleccionado?')){
      onEliminar(pasaporte)
    }
    
  };
  return (
    <Button onClick={handleEliminar}>
      <span role="img" aria-labelledby="trash">
        üóëÔ∏è
      </span>
    </Button>
  );
};

BtnEliminarPasaporte.propTypes = {
  pasaporte: PropTypes.shape({
    id: PropTypes.number.isRequired,
    fullname: PropTypes.string.isRequired,
    address: PropTypes.string.isRequired,
    birthdate: PropTypes.string.isRequired,
    document_id: PropTypes.number.isRequired,
    application_date: PropTypes.string.isRequired,
    appointment_date: PropTypes.string.isRequired,
    amount: PropTypes.number.isRequired,
  }).isRequired,
  onEliminar: PropTypes.func.isRequired,
};
