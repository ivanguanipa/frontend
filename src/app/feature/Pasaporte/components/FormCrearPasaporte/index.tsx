import * as PropTypes from 'prop-types';
import * as React from 'react';
import * as Yup from 'yup';
import { Button } from 'app/shared/components/Button';
import { FormikHelpers } from 'formik/dist/types';
import { Input } from 'app/shared/components/Input';
import { Pasaporte } from '../../models/Pasaporte';
import { SpanError } from './styles';
import { useFormik } from 'formik';

type Nullable<T> = T | null ;

interface FormValues {
  fullname: string;
  address: string;
  birthdate: string;
  documentId: number;
  applicationDate: string;
  appointmentDate?:  Nullable<string>;
  id?:  Nullable<number>;
  amount?:  Nullable<number>;
}

interface FormCrearPasaporteProp {
  onSubmit: (payload: Pasaporte) => any;
  disabled?: boolean;
  formTitle: string;
  initialValues?: FormValues;
}

const validationSchema = Yup.object().shape<FormValues>({
  fullname: Yup.string().required('El campo nombre y apellido es requerido.'),
  address: Yup.string().required('El campo dirección es requerido.'),
  birthdate: Yup.string().required('El campo fecha de nacimiento es requerido.'),
  documentId: Yup.number().moreThan(0,'El campo cédula es requerido.').required('El campo cédula es requerido.'),
  applicationDate: Yup.string().required('El campo fecha de solicitud es requerido.'),
  appointmentDate: Yup.string().nullable(),
  amount: Yup.number().nullable(),
  id: Yup.number().nullable(),
});

export const FormCrearPasaporte: React.FC<FormCrearPasaporteProp> = ({
  onSubmit,
  disabled,
  formTitle,
  initialValues = {
    fullname:'',
    address: '',
    birthdate: '',
    documentId: 0,
    applicationDate: '',
    appointmentDate: null,
    amount: null,
    id: null,
  },
}) => {
  const handleSubmit = (
    values: FormValues,
    { resetForm }: FormikHelpers<FormValues>
  ) => {
    onSubmit(values);
    resetForm();
  };
  const formik = useFormik({
    initialValues,
    validationSchema,
    onSubmit: handleSubmit,
  });

  return (
    <form onSubmit={formik.handleSubmit}>
      <h2 style={{marginBottom:'15px'}}>{formTitle}</h2>
      <div><label htmlFor="fullname">Nombre y Apellido</label></div>
      <Input
        disabled={disabled}
        name="fullname"
        placeholder="Nombre y Apellido"
        value={formik.values.fullname}
        onChange={formik.handleChange}
      />
      {formik.errors.fullname && (
        <SpanError>{formik.errors.fullname}</SpanError>
      )}
      <div> <label htmlFor="fullname">Dirección</label></div>
      <Input
        disabled={disabled}
        name="address"
        placeholder="Dirección"
        value={formik.values.address}
        onChange={formik.handleChange}
      />
      {formik.errors.address && (
        <SpanError>{formik.errors.address}</SpanError>
      )}
      <div><label htmlFor="fullname">Fecha de Nacimiento</label></div>
      <Input
      type='date'
        disabled={disabled}
        name="birthdate"
        placeholder="Fecha de Nacimiento"
        value={formik.values.birthdate}
        onChange={formik.handleChange}
      />
      {formik.touched.birthdate && formik.errors.birthdate && (
        <SpanError>{formik.errors.birthdate}</SpanError>
      )}
      <div><label htmlFor="fullname">Cédula</label></div>
      <Input
      type='number'
        disabled={disabled}
        name="documentId"
        placeholder="Cédula"
        value={formik.values.documentId===0?'':formik.values.documentId}
        onChange={formik.handleChange}
      />
      {formik.touched.documentId && formik.errors.documentId && (
        <SpanError>{formik.errors.documentId}</SpanError>
      )}
      <div><label htmlFor="fullname">Fecha de la Solicitud</label></div>

      <Input
        type='date'
        disabled={disabled}
        name="applicationDate"
        placeholder="Fecha de la Solicitud"
        value={formik.values.applicationDate}
        onChange={formik.handleChange}
      />
      {formik.errors.applicationDate && (
        <SpanError>{formik.errors.applicationDate}</SpanError>
      )}
      <Input
        type='hidden'
        disabled={disabled}
        name="appointmentDate"
        value={formik.values.applicationDate}
      />
      <Input
        type='hidden'
        disabled={disabled}
        name="amount"
        value={formik.values.applicationDate}

      />
     
      <Button type="submit">Registrar</Button>
    </form>
  );
};

FormCrearPasaporte.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  formTitle: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  initialValues: PropTypes.shape({
    fullname: PropTypes.string.isRequired,
    address: PropTypes.string.isRequired,
    birthdate: PropTypes.string.isRequired,
    documentId: PropTypes.number.isRequired,
    applicationDate: PropTypes.string.isRequired,
    appointmentDate: PropTypes.string,
    amount: PropTypes.number,
  }),
};
