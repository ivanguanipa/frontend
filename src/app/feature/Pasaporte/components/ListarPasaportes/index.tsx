import * as PropTypes from 'prop-types';
import * as React from 'react';
import { BtnEliminarPasaporte } from '../EliminarPasaporte';
import { BtnMostrarPasaporte } from '../BtnMostrarPasaporte';
import { Pasaporte } from '../../models/Pasaporte';
import { Table } from './styles';
import {dateFormat} from "app/utils/index";

export interface ListaPasaportesProps {
  pasaportes: Array<Pasaporte>;
  onClickEliminarPasaporte: (pasaporte: Pasaporte) => any;
}

export const ListaPasaportes: React.FC<ListaPasaportesProps> = ({
  pasaportes,
  onClickEliminarPasaporte,
}) => {
  return (
    <Table>
      <thead>
        <tr>
          <td>
            <b>Nombre y Apellido</b>
          </td>
          <td>
            <b>Cedula</b>
          </td>
          <td>
            <b>Dirección</b>
          </td>
          <td>
            <b>Fecha de Solicitud</b>
          </td>
          <td>
            <b>Fecha de la Cita</b>
          </td>
          <td>
            <b>Monto</b>
          </td>
          <td>
            <b>Acción</b>
          </td>
        </tr>
      </thead>
      <tbody>
        {pasaportes.map((pasaporte: Pasaporte) => {
          return (
            <tr key={pasaporte.document_id}>
              <td>{pasaporte.fullname}</td>
              <td>{pasaporte.document_id}</td>
              <td>{pasaporte.address}</td>
              <td>{dateFormat(pasaporte.application_date,"DD-MM-YYYY")}</td>
              <td>{dateFormat(String(pasaporte.appointment_date),"DD-MM-YYYY")}</td>
              <td>{pasaporte.amount}</td>
              
              <td>
                <BtnMostrarPasaporte
                  pasaporte={pasaporte}
                ></BtnMostrarPasaporte>
                <BtnEliminarPasaporte
                  pasaporte={pasaporte}
                  onEliminar={()=>onClickEliminarPasaporte(pasaporte)}
                ></BtnEliminarPasaporte>
              </td>
            </tr>
          );
        })}
      </tbody>
    </Table>
  );
};

ListaPasaportes.propTypes = {
  pasaportes: PropTypes.array.isRequired,
  onClickEliminarPasaporte: PropTypes.func.isRequired,
};
