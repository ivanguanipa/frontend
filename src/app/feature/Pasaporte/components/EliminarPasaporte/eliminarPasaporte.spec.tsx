import * as React from 'react';
import { ShallowWrapper, shallow } from 'enzyme';
import { BtnEliminarPasaporte } from 'app/feature/Pasaporte/components/EliminarPasaporte';
import { Pasaporte } from 'app/feature/Pasaporte/models/Pasaporte';
import { Button } from 'app/shared/components/Button';

describe('Btn Eliminar Pasaportes Test', () => {
  let componentWrapper: ShallowWrapper;

  afterEach(() => {
    componentWrapper.unmount();
  });

  it('should match snapshot', () => {
    const data: Pasaporte = {
        id: 1,
        fullname: "string",
        address: "string",
        birthdate: "string",
        documentId:1,
        applicationDate: "string",
        appointmentDate: "string",
        amount: 1,
    };

    componentWrapper = shallow(
      <BtnEliminarPasaporte
      pasaporte={data}
        onEliminar={() => {}}
      />
    );
    expect(componentWrapper).toMatchSnapshot();
  });

  it('test eliminar pasaporte',()=>{
    const mockEliminar = jest.fn();
    spyOn(window,'confirm').and.returnValue(true);
    const data: Pasaporte = {
        id: 1,
        fullname: "string",
        address: "string",
        birthdate: "string",
        documentId:1,
        applicationDate: "string",
        appointmentDate: "string",
        amount: 1,
    };

    componentWrapper = shallow(
        <BtnEliminarPasaporte
      pasaporte={data}
        onEliminar={()=>mockEliminar(1)}
      />     
      );

    const buttons = componentWrapper.find('.deletePassport');

   // spyOn(componentWrapper.props(),'mockEliminar').and.returnValue(true);

    buttons.simulate('click');
    expect(mockEliminar).toHaveBeenCalled();
  })
  it('test negar eliminar pasaporte',()=>{
    const mockEliminar = jest.fn();
    spyOn(window,'confirm').and.returnValue(false);
    const data: Pasaporte = {
        id: 1,
        fullname: "string",
        address: "string",
        birthdate: "string",
        documentId:1,
        applicationDate: "string",
        appointmentDate: "string",
        amount: 1,
    };

    componentWrapper = shallow(
        <BtnEliminarPasaporte
      pasaporte={data}
        onEliminar={()=>mockEliminar(1)}
      />     
      );

    const buttons = componentWrapper.find('.deletePassport');

   // spyOn(componentWrapper.props(),'mockEliminar').and.returnValue(true);

    buttons.simulate('click');
    expect(mockEliminar).not.toHaveBeenCalled();
  })
 
});

export {};
