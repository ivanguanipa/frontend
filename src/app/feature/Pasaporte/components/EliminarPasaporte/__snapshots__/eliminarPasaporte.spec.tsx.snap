// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn Eliminar Pasaportes Test should match snapshot 1`] = `
<styled.button
  className="deletePassport"
  onClick={[Function]}
>
  <span
    aria-labelledby="trash"
    role="img"
  >
    🗑️
  </span>
</styled.button>
`;
